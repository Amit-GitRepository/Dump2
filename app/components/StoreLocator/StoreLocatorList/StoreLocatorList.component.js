import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './StoreLocatorList.style';
import trueShopMapViewTestIDs from '../../../config/testid/trueShopMapViewPage';
import {InfoBar, List} from '../../Shared';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

class StoreLocatorList extends Component {

  render () {
    const {onStorePress, storeList, radiusKM} = this.props;
    const infoBarText = translate('STORE_LOCATOR__COVERAGE', {distance: radiusKM});
    return (<View style={styles.container}>
      <View style={styles.wrapper}>
        <InfoBar text={infoBarText} accessibilityLabel={trueShopMapViewTestIDs.LABEL__INFOBAR_COVERAGE} testID={trueShopMapViewTestIDs.LABEL__INFOBAR_COVERAGE}/>
        <List onPress={onStorePress} data={storeList} headingKey='storeName' subHeadingKey='subheading'/>
      </View>
    </View>);
  }
}

StoreLocatorList.defaultProps = {
  onStorePress: noop,
  storeList: []
};

StoreLocatorList.propTypes = {
  onStorePress: PropTypes.func,
  storeList: PropTypes.array,
  radiusKM: PropTypes.number
};

export default StoreLocatorList;
