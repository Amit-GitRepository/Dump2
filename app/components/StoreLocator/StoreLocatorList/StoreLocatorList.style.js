export default {
  container: {
    flex: 1
  },
  wrapper: {
    width: '100%',
    position: 'absolute',
    top: 0,
    bottom: -4
  }
};
