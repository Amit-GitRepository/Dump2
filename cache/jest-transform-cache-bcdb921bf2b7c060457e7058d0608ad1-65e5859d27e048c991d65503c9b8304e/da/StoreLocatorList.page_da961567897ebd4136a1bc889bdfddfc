Object.defineProperty(exports,"__esModule",{value:true});exports.mapDispatchToProps=exports.mapStateToProps=undefined;var cov_2kjf7qmvlj=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/pages/StoreLocator/StoreLocatorList.page.js',hash='e36fe9ea8210c8841d6f0b56d7b02fcfc6bd4b70',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/pages/StoreLocator/StoreLocatorList.page.js',statementMap:{'0':{start:{line:12,column:51},end:{line:12,column:61}},'1':{start:{line:13,column:22},end:{line:16,column:6}},'2':{start:{line:14,column:6},end:{line:14,column:65}},'3':{start:{line:15,column:6},end:{line:15,column:19}},'4':{start:{line:17,column:4},end:{line:18,column:58}},'5':{start:{line:22,column:0},end:{line:24,column:2}},'6':{start:{line:25,column:0},end:{line:29,column:2}},'7':{start:{line:31,column:31},end:{line:38,column:1}},'8':{start:{line:32,column:23},end:{line:32,column:69}},'9':{start:{line:33,column:22},end:{line:35,column:147}},'10':{start:{line:33,column:34},end:{line:35,column:146}},'11':{start:{line:36,column:31},end:{line:36,column:60}},'12':{start:{line:37,column:2},end:{line:37,column:77}},'13':{start:{line:40,column:34},end:{line:42,column:2}},'14':{start:{line:40,column:49},end:{line:42,column:1}},'15':{start:{line:41,column:27},end:{line:41,column:113}},'16':{start:{line:41,column:33},end:{line:41,column:113}},'17':{start:{line:44,column:38},end:{line:44,column:104}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:11,column:2},end:{line:11,column:3}},loc:{start:{line:11,column:19},end:{line:19,column:3}},line:11},'1':{name:'(anonymous_1)',decl:{start:{line:13,column:39},end:{line:13,column:40}},loc:{start:{line:13,column:50},end:{line:16,column:5}},line:13},'2':{name:'(anonymous_2)',decl:{start:{line:31,column:31},end:{line:31,column:32}},loc:{start:{line:31,column:42},end:{line:38,column:1}},line:31},'3':{name:'(anonymous_3)',decl:{start:{line:33,column:22},end:{line:33,column:23}},loc:{start:{line:33,column:34},end:{line:35,column:146}},line:33},'4':{name:'(anonymous_4)',decl:{start:{line:40,column:34},end:{line:40,column:35}},loc:{start:{line:40,column:49},end:{line:42,column:1}},line:40},'5':{name:'(anonymous_5)',decl:{start:{line:41,column:16},end:{line:41,column:17}},loc:{start:{line:41,column:27},end:{line:41,column:113}},line:41},'6':{name:'(anonymous_6)',decl:{start:{line:41,column:27},end:{line:41,column:28}},loc:{start:{line:41,column:33},end:{line:41,column:113}},line:41}},branchMap:{'0':{loc:{start:{line:14,column:24},end:{line:14,column:64}},type:'binary-expr',locations:[{start:{line:14,column:24},end:{line:14,column:44}},{start:{line:14,column:48},end:{line:14,column:64}}],line:14}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0,'15':0,'16':0,'17':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _ComponentWithTab2=require('../../components/ComponentWithTab/ComponentWithTab.component');var _ComponentWithTab3=_interopRequireDefault(_ComponentWithTab2);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _result=require('lodash/result');var _result2=_interopRequireDefault(_result);
var _reactRedux=require('react-redux');
var _Shared=require('../../components/Shared');
var _reactNavigation=require('react-navigation');
var _helper=require('../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

StoreLocatorListPage=function(_ComponentWithTab){_inherits(StoreLocatorListPage,_ComponentWithTab);function StoreLocatorListPage(){_classCallCheck(this,StoreLocatorListPage);return _possibleConstructorReturn(this,(StoreLocatorListPage.__proto__||Object.getPrototypeOf(StoreLocatorListPage)).apply(this,arguments));}_createClass(StoreLocatorListPage,[{key:'renderWithTab',value:function renderWithTab()
{cov_2kjf7qmvlj.f[0]++;var _ref=(cov_2kjf7qmvlj.s[0]++,
this.props),nearByStores=_ref.nearByStores,onStorePress=_ref.onStorePress,language=_ref.language;
var storeList=(cov_2kjf7qmvlj.s[1]++,nearByStores.map(function(store){cov_2kjf7qmvlj.f[1]++;cov_2kjf7qmvlj.s[2]++;
store.storeName=(cov_2kjf7qmvlj.b[0][0]++,store.name[language])||(cov_2kjf7qmvlj.b[0][1]++,store.name['en']);cov_2kjf7qmvlj.s[3]++;
return store;
}));cov_2kjf7qmvlj.s[4]++;
return _react2.default.createElement(_Shared.List,{onPress:onStorePress,data:storeList,
headingKey:'storeName',subHeadingKey:'subheading'});
}}]);return StoreLocatorListPage;}(_ComponentWithTab3.default);cov_2kjf7qmvlj.s[5]++;


StoreLocatorListPage.defaultProps={
nearByStores:null};cov_2kjf7qmvlj.s[6]++;

StoreLocatorListPage.propTypes={
onStorePress:_propTypes2.default.func,
nearByStores:_propTypes2.default.array,
language:_propTypes2.default.string};cov_2kjf7qmvlj.s[7]++;


var mapStateToProps=exports.mapStateToProps=function mapStateToProps(state){cov_2kjf7qmvlj.f[2]++;
var nearByStores=(cov_2kjf7qmvlj.s[8]++,(0,_result2.default)(state,'storeLocator.nearByStores',[]));cov_2kjf7qmvlj.s[9]++;
var addDistance=function addDistance(store){cov_2kjf7qmvlj.f[3]++;cov_2kjf7qmvlj.s[10]++;return _extends({},
store,{
subheading:(0,_helper.translate)('STORE_LOCATOR__DETAILS_DISTANCE')+' '+(0,_helper.translate)('STORE_LOCATOR__DETAILS_DISTANCE_VAL',{distance:store.distance})});};
var storesWithSubheading=(cov_2kjf7qmvlj.s[11]++,nearByStores.map(addDistance));cov_2kjf7qmvlj.s[12]++;
return{nearByStores:storesWithSubheading,language:state.user.language};
};cov_2kjf7qmvlj.s[13]++;

var mapDispatchToProps=exports.mapDispatchToProps=function mapDispatchToProps(dispatch){cov_2kjf7qmvlj.f[4]++;cov_2kjf7qmvlj.s[14]++;return{
onStorePress:function onStorePress(store){cov_2kjf7qmvlj.f[5]++;cov_2kjf7qmvlj.s[15]++;return function(){cov_2kjf7qmvlj.f[6]++;cov_2kjf7qmvlj.s[16]++;return dispatch(_reactNavigation.NavigationActions.navigate({routeName:'StoreDetails',params:store}));};}};
};

var ConnectedStoreLocatorListPage=(cov_2kjf7qmvlj.s[17]++,(0,_reactRedux.connect)(mapStateToProps,mapDispatchToProps)(StoreLocatorListPage));exports.default=
ConnectedStoreLocatorListPage;