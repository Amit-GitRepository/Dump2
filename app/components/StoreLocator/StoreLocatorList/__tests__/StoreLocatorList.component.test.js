import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import StoreLocatorList from '../StoreLocatorList.component';

describe('StoreLocatorList component', () => {
  it('should render correctly', () => {
    const componentTree = renderer.create(<StoreLocatorList />).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
