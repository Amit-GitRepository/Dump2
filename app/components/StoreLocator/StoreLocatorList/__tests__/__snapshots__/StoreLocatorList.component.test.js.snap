// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocatorList component should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": -4,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="trueShopMapView_labelInfobar_coverage"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(40, 60, 70, 0.8)",
          },
          Object {},
        ]
      }
      testID="trueShopMapView_labelInfobar_coverage"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFFFFF",
            "fontFamily": "SukhumvitSet-Text",
            "fontSize": 18,
            "padding": 2,
            "textAlign": "center",
          }
        }
      >
        Coverage [missing {{distance}} value] k.m.
      </Text>
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
